### ISSUE TITLE
Better handling for deferred types once data analysis work is complete

### ISSUE THREAD

**Issue by MikeStall** [microsoft/Power-Fx#837]
Regarding deferred type support - currently wraps CheckInvocation, but that is going away. 

In the future DataFlow analysis will only call CheckTypes and never call CheckSemantics
CheckSemantics would happen way later at bind time

Check with Bryan for more details.
